src/include/api_trace.h

```diff
@@ -0,0 +1,197 @@
+    ncclComm_t comm, cudaStream_t stream);
+
+ncclResult_t ncclRecv_impl(void* recvbuff, size_t count, ncclDataType_t datatype, int peer,
+    ncclComm_t comm, cudaStream_t stream);
+
+namespace rccl {
+    typedef uint64_t rccl_range_id_t;
+    typedef ncclResult_t (*ncclAllGatherfn_t)(const void* sendbuff, void* recvbuff, size_t sendcount,
+        ncclDataType_t datatype, ncclComm_t comm, hipStream_t stream);
+    typedef ncclResult_t (*ncclAllReduce_fn_t)(const void* sendbuff, void* recvbuff, size_t count,
+        ncclDataType_t datatype, ncclRedOp_t op, ncclComm* comm, cudaStream_t stream);
+    typedef ncclResult_t (*ncclAllToAll_fn_t)(const void* sendbuff, void* recvbuff, size_t count,
+        ncclDataType_t datatype, ncclComm_t comm, hipStream_t stream);
+    typedef ncclResult_t (*ncclAllToAllv_fn_t)(const void *sendbuff, const size_t sendcounts[], const size_t sdispls[],
+        void *recvbuff, const size_t recvcounts[], const size_t rdispls[],
+        ncclDataType_t datatype, ncclComm_t comm, hipStream_t stream);
+    typedef ncclResult_t (*ncclBroadcast_fn_t)(const void* sendbuff, void* recvbuff, size_t count, ncclDataType_t datatype, int root,
+        ncclComm_t comm, cudaStream_t stream);
+    typedef ncclResult_t (*ncclGather_fn_t)(const void* sendbuff, void* recvbuff, size_t sendcount,
+        ncclDataType_t datatype, int root, ncclComm_t comm, hipStream_t stream);
+    typedef ncclResult_t (*ncclReduce_fn_t)(const void* sendbuff, void* recvbuff, size_t count,
+        ncclDataType_t datatype, ncclRedOp_t op, int root, ncclComm_t comm, cudaStream_t stream);
+    typedef ncclResult_t (*ncclReduceScatter_fn_t)(const void* sendbuff, void* recvbuff, size_t recvcount,
+        ncclDataType_t datatype, ncclRedOp_t op, ncclComm* comm, cudaStream_t stream);
+    typedef ncclResult_t (*ncclScatter_fn_t)(const void* sendbuff, void* recvbuff, size_t recvcount, ncclDataType_t datatype, int root,
+        ncclComm_t comm, hipStream_t stream);
+    typedef ncclResult_t (*ncclSend_fn_t)(const void* sendbuff, size_t count, ncclDataType_t datatype, int peer,
+        ncclComm_t comm, cudaStream_t stream);
+    typedef ncclResult_t (*ncclRecv_fn_t)(void* recvbuff, size_t count, ncclDataType_t datatype, int peer,
+        ncclComm_t comm, cudaStream_t stream);
+
+
+    struct rcclApiFuncTable
+    {
+        uint64_t               size;
+        ncclAllGatherfn_t      ncclAllGather_fn;
+        ncclAllReduce_fn_t     ncclAllReduce_fn;
+        ncclAllToAll_fn_t      ncclAllToAll_fn;
+        ncclAllToAllv_fn_t     ncclAllToAllv_fn;
+        ncclBroadcast_fn_t     ncclBroadcast_fn;
+        ncclGather_fn_t        ncclGather_fn;
+        ncclReduce_fn_t        ncclReduce_fn;
+        ncclReduceScatter_fn_t ncclReduceScatter_fn;
+        ncclScatter_fn_t       ncclScatter_fn;
+        ncclSend_fn_t          ncclSend_fn;
+        ncclRecv_fn_t          ncclRecv_fn;
+    };
```

Comment:
This should be the bulk of the header. Just about everything else should be in an implementation file.

Reply:
In addition to the above, we need the `RCCL_API_TABLE_MAJOR_VERSION` and `RCCL_API_TABLE_PATCH_VERSION defines in this header.